File: CS/BatchEditEditItemAndDataItemTemplates/App_Code/DataModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

public static class DataModel {
    private const string ProductsSessionKey = "products";

    public static List<Product> GetProducts() {
        if (HttpContext.Current.Session[ProductsSessionKey] == null)
            GenerateProducts();
        return (List<Product>)HttpContext.Current.Session[ProductsSessionKey];
    }
    public static void UpdateProducts(Product product) {
        List<Product> products = GetProducts();
        int index = products.FindIndex(p => p.ID == product.ID);
        products[index] = product;
        SaveProducts(products);
    }
    public static void InsertProduct(Product product) {
        List<Product> products = GetProducts();
        product.ID = products.Count;
        products.Add(product);
        SaveProducts(products);
    }
    public static void RemoveProduct(Product product) {
        List<Product> products = GetProducts();
        products.RemoveAll(p => p.ID == product.ID);
        SaveProducts(products);
    }
    public static void GenerateProducts() {
        List<Product> products = new List<Product>();
        for (int j = 0; j < 10; j++) {
            for (int i = 0; i < 5; i++) {
                int ratingValue = j % 2 == 0 ? i + 1 : 5 - i;
                products.Add(new Product() { ID = j * 5 + i, RatingValue = ratingValue });
            }
        }
        HttpContext.Current.Session[ProductsSessionKey] = products;
    }
    private static void SaveProducts(List<Product> products) {
        HttpContext.Current.Session[ProductsSessionKey] = products;
    }
}
public class Product {
    public int ID { get; set; }
    public int RatingValue { get; set; }
}

File: VB/BatchEditEditItemAndDataItemTemplates/App_Code/DataModel.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web

Public NotInheritable Class DataModel

	Private Sub New()
	End Sub
	Private Const ProductsSessionKey As String = "products"

	Public Shared Function GetProducts() As List(Of Product)
		If HttpContext.Current.Session(ProductsSessionKey) Is Nothing Then
			GenerateProducts()
		End If
		Return CType(HttpContext.Current.Session(ProductsSessionKey), List(Of Product))
	End Function
	Public Shared Sub UpdateProducts(ByVal product As Product)
		Dim products As List(Of Product) = GetProducts()
		Dim index As Integer = products.FindIndex(Function(p) p.ID = product.ID)
		products(index) = product
		SaveProducts(products)
	End Sub
	Public Shared Sub InsertProduct(ByVal product As Product)
		Dim products As List(Of Product) = GetProducts()
		product.ID = products.Count
		products.Add(product)
		SaveProducts(products)
	End Sub
	Public Shared Sub RemoveProduct(ByVal product As Product)
		Dim products As List(Of Product) = GetProducts()
		products.RemoveAll(Function(p) p.ID = product.ID)
		SaveProducts(products)
	End Sub
	Public Shared Sub GenerateProducts()
		Dim products As New List(Of Product)()
		For j As Integer = 0 To 9
			For i As Integer = 0 To 4
				Dim ratingValue As Integer = If(j Mod 2 = 0, i + 1, 5 - i)
				products.Add(New Product() With {.ID = j * 5 + i, .RatingValue = ratingValue})
			Next i
		Next j
		HttpContext.Current.Session(ProductsSessionKey) = products
	End Sub
	Private Shared Sub SaveProducts(ByVal products As List(Of Product))
		HttpContext.Current.Session(ProductsSessionKey) = products
	End Sub
End Class
Public Class Product
	Public Property ID() As Integer
	Public Property RatingValue() As Integer
End Class